### ISSUE TITLE
docker start failed with 404 page not found

### ISSUE THREAD

**Issue by jacknlliu** [weaveworks/weave#3054]
I test the following commands,
```
$ weave launch --no-dns

$ eval $(weave env)
```

then

```
$ docker run -it ubuntu:16.04 bash

docker: Error response from daemon: unable to register f8fe8aea2f12634faa6676b6605e124c678067410c47063f41bff61a4ff3d841 with weaveDNS: 404 Not Found: 404 page not found
Error: failed to start containers
```

## Versions:
```
$ weave version
2.0.1

$ docker version
Client:
 Version:      17.06.0-ce
 API version:  1.30
 Go version:   go1.8.3
 Git commit:   02c1d87
 Built:        Fri Jun 23 21:22:33 2017
 OS/Arch:      linux/amd64

Server:
 Version:      17.06.0-ce
 API version:  1.30 (minimum version 1.12)
 Go version:   go1.8.3
 Git commit:   02c1d87
 Built:        Fri Jun 23 21:23:51 2017
 OS/Arch:      linux/amd64
 Experimental: false

$ uname -a
Linux  4.11.8-200.fc25.x86_64 #1 SMP Thu Jun 29 16:13:56 UTC 2017 x86_64 x86_64 x86_64 GNU/Linux
```

## Logs:
```
$ docker logs weave

INFO: 2017/07/10 03:04:09.969608 GET /_ping
INFO: 2017/07/10 03:04:09.970265 GET /v1.30/images/json
INFO: 2017/07/10 03:04:22.507601 GET /_ping
INFO: 2017/07/10 03:04:22.514763 POST /v1.30/containers/create
INFO: 2017/07/10 03:04:22.514932 Creating container with WEAVE_CIDR ""
INFO: 2017/07/10 03:04:25.276593 POST /v1.30/containers/f8fe8aea2f12634faa6676b6605e124c678067410c47063f41bff61a4ff3d841/attach?stderr=1&stdin=1&stdout=1&stream=1
INFO: 2017/07/10 03:04:25.279439 POST /v1.30/containers/f8fe8aea2f12634faa6676b6605e124c678067410c47063f41bff61a4ff3d841/wait?condition=next-exit
INFO: 2017/07/10 03:04:25.281895 POST /v1.30/containers/f8fe8aea2f12634faa6676b6605e124c678067410c47063f41bff61a4ff3d841/start
INFO: 2017/07/10 03:04:25.957226 Attaching container f8fe8aea2f12634faa6676b6605e124c678067410c47063f41bff61a4ff3d841 with WEAVE_CIDR "" to weave network
INFO: 2017/07/10 03:04:26.052918 Discovered local MAC 7a:db:b4:aa:06:69
WARN: 2017/07/10 03:04:26.121546 read unix /var/run/weave/weave.sock->@: use of closed network connection
WARN: 2017/07/10 03:04:27.116142 Error intercepting response: unable to register f8fe8aea2f12634faa6676b6605e124c678067410c47063f41bff61a4ff3d841 with weaveDNS: 404 Not Found: 404 page not found
: 404 Not Found: 404 page not found
```

**Comment by marccarre** [comment]
Hi @jacknlliu, 
Many thanks for raising this issue and the detailed steps & symptoms
I confirm I can definitely reproduce the issue :+1:

**Comment by marccarre** [comment]
@jacknlliu, just to better understand your use case, and advise in case there is a workaround:
Why do you need to disable the DNS at the moment?

**Comment by jacknlliu** [comment]
@marccarre yeah, I disabled DNS because I also use `weave` on a ARM device which will show the following error if not disable DNS,
```
FATA: 2017/07/09 16:18:27.336931 Unable to start dns server: listen udp 172.17.0.1:53: bind: address already in use
```
and clock skew issue is also confusing.

**Comment by phreax** [comment]
@jacknlliu I had the exact same issues, as I don't want to use `weave-dns`. When I tried to launch docker container using the weave socket I would get the same `404` error for `weaveDNS`.

After reading the [docs](https://github.com/weaveworks/weave/blob/master/site/weavedns.md) again, the solution was simple though not immediate obvious:

> To disable an application container's use of weaveDNS, add the `--without-dns` option to weave launch. To disable weaveDNS itself, launch weave with the `--no-dns` option.

So you actually need to apply both options to launch weave, in order do disable the DNS server (which is useful, if you have another DNS server running) and make the containers not use this DNS server:

```
$ weave launch --no-dns --without-dns
```

Cheers 😄

**Comment by jacknlliu** [comment]
Thanks for your reply, I also think that DNS server is not needed for my use case.

**Comment by bboreham** [comment]
Fixed by #3088
